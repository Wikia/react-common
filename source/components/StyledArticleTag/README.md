Defaults:

```js
<StyledArticleTag tag="This is a default tag" />
```

With extra options

```js
<StyledArticleTag
    tag="This is a secondary/removable tag"
    removable
    secondary
    onRemove={() => console.log('Remove tag clicked')}
    onClick={() => console.log('Tag clicked')}
/>
```
