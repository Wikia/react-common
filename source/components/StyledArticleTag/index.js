import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import IconClose from '../../icons/IconCloseSmall';

const Wrapper = styled.span`
    align-items: center;
    border-radius: 25px;
    border: 1px solid ${({ theme }) => theme.color.black};
    cursor: pointer;
    display: inline-flex;
    font-size: ${({ theme }) => theme.font_size.s};
    line-height:  ${({ theme }) => theme.line_height.tight};
    padding: 2px 12px;

    ${({ secondary }) => secondary && css`
        background-color: ${({ theme }) => theme.color.fandom_light_gray};
        border: none;
    `}
`;

const Icon = styled(IconClose)`
    cursor: pointer;
    height: 11px;
    margin-left: 5px;
    width: 11px;
`;

function StyledArticleTag({
    className,
    onClick,
    onRemove,
    removable,
    secondary,
    tag,
}) {
    return (
        <Wrapper onClick={onClick} className={className} secondary={secondary}>
            {tag}
            {removable && <Icon onClick={onRemove} />}
        </Wrapper>
    );
}

StyledArticleTag.propTypes = {
    /** Extra class name */
    className: PropTypes.string,
    onClick: PropTypes.func,
    onRemove: PropTypes.func,
    /** Flag to mark tag as removable */
    removable: PropTypes.bool,
    /** Secondary flag */
    secondary: PropTypes.bool,
    /** Tag name */
    tag: PropTypes.string.isRequired,
};

StyledArticleTag.defaultProps = {
    className: '',
    onClick: null,
    onRemove: null,
    removable: false,
    secondary: false,
};

export default StyledArticleTag;
