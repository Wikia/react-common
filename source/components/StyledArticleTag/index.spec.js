import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import StyledArticleTag from './index';

/* eslint-disable no-alert */

test('StyledArticleTag renders correctly with default values', () => {
    const component = mountWithThemeProvider(
        <StyledArticleTag tag="foo" />,
    ).find(StyledArticleTag);

    expect(component).toMatchSnapshot();
});

test('StyledArticleTag renders correctly as secondary and removable', () => {
    const component = mountWithThemeProvider(
        <StyledArticleTag tag="foo" secondary removable />,
    ).find(StyledArticleTag);

    expect(component).toMatchSnapshot();
});

test('StyledArticleTag onClick is clickable', () => {
    const mockOnClick = sinon.spy();
    const wrapper = mountWithThemeProvider(
        <StyledArticleTag tag="foo" onClick={mockOnClick} />,
    ).find(StyledArticleTag);

    wrapper.find('span').simulate('click');

    expect(mockOnClick.calledOnce).toBe(true);
});

test('StyledArticleTag onRemove is clickable', () => {
    const onRemove = sinon.spy();
    const wrapper = mountWithThemeProvider(
        <StyledArticleTag tag="foo" onRemove={onRemove} removable />,
    ).find(StyledArticleTag);

    wrapper.find('StyledArticleTag__Icon').simulate('click');

    expect(onRemove.calledOnce).toBe(true);
});

// test('Button triggers onClick event', () => {
//     const mockOnClick = sinon.spy();
//     const wrapper = shallow(
//         <Button onClick={mockOnClick}>clickable</Button>
//     );

//     wrapper.find('button').simulate('click');

//     expect(mockOnClick.calledOnce).toBe(true);
// });

// test('Button renders correctly with custom types values', () => {
//     let component = renderer.create(
//         <Button disabled>disabled</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button secondary>secondary</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button secondary light>secondary light</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button text>text</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button text light>text light</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button square>&hellip;</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();

//     component = renderer.create(
//         <Button className="custom-class">with custom class name</Button>,
//     );
//     expect(component.toJSON()).toMatchSnapshot();
// });
