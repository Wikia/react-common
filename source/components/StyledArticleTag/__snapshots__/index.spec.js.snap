// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledArticleTag renders correctly as secondary and removable 1`] = `
<StyledArticleTag
  className=""
  onClick={null}
  onRemove={null}
  removable={true}
  secondary={true}
  tag="foo"
>
  <StyledArticleTag__Wrapper
    className=""
    onClick={null}
    secondary={true}
  >
    <StyledComponent
      className=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "StyledArticleTag__Wrapper-gxea7p-0",
            "isStatic": false,
            "lastClassName": "iMfhyl",
            "rules": Array [
              "align-items:center;border-radius:25px;border:1px solid ",
              [Function],
              ";cursor:pointer;display:inline-flex;font-size:",
              [Function],
              ";line-height:",
              [Function],
              ";padding:2px 12px;",
              [Function],
            ],
          },
          "displayName": "StyledArticleTag__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "StyledArticleTag__Wrapper-gxea7p-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={null}
      secondary={true}
    >
      <span
        className="StyledArticleTag__Wrapper-gxea7p-0 iMfhyl"
        onClick={null}
      >
        foo
        <StyledArticleTag__Icon
          onClick={null}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "StyledArticleTag__Icon-gxea7p-1",
                  "isStatic": true,
                  "lastClassName": "bHZlOu",
                  "rules": Array [
                    "cursor:pointer;height:11px;margin-left:5px;width:11px;",
                  ],
                },
                "displayName": "StyledArticleTag__Icon",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "StyledArticleTag__Icon-gxea7p-1",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={null}
          >
            <_default
              className="StyledArticleTag__Icon-gxea7p-1 bHZlOu"
              onClick={null}
            >
              <_default
                className="StyledArticleTag__Icon-gxea7p-1 bHZlOu"
                onClick={null}
              >
                <div
                  className="StyledArticleTag__Icon-gxea7p-1 bHZlOu"
                  onClick={null}
                />
              </_default>
            </_default>
          </StyledComponent>
        </StyledArticleTag__Icon>
      </span>
    </StyledComponent>
  </StyledArticleTag__Wrapper>
</StyledArticleTag>
`;

exports[`StyledArticleTag renders correctly with default values 1`] = `
<StyledArticleTag
  className=""
  onClick={null}
  onRemove={null}
  removable={false}
  secondary={false}
  tag="foo"
>
  <StyledArticleTag__Wrapper
    className=""
    onClick={null}
    secondary={false}
  >
    <StyledComponent
      className=""
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "StyledArticleTag__Wrapper-gxea7p-0",
            "isStatic": false,
            "lastClassName": "iTCycT",
            "rules": Array [
              "align-items:center;border-radius:25px;border:1px solid ",
              [Function],
              ";cursor:pointer;display:inline-flex;font-size:",
              [Function],
              ";line-height:",
              [Function],
              ";padding:2px 12px;",
              [Function],
            ],
          },
          "displayName": "StyledArticleTag__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "StyledArticleTag__Wrapper-gxea7p-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={null}
      secondary={false}
    >
      <span
        className="StyledArticleTag__Wrapper-gxea7p-0 iTCycT"
        onClick={null}
      >
        foo
      </span>
    </StyledComponent>
  </StyledArticleTag__Wrapper>
</StyledArticleTag>
`;
